app/controllers/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -78,6 +78,29 @@ def reassign_hostgroup
         end
       end
 
+      api :PUT, "/hosts/bulk/assign_taxonomy", N_("Assign organization or location")
```

Comment:
I think the whole method will be easier if you rewrite this to separate API paths. So `assign_organization` and `assign_location`. Most lines within this are simply there to support 2 different types, but as you can see in the translations that very quickly gets messy.